augur/util_support/node_data_file.py

```diff
@@ -103,3 +103,17 @@ def validate(self):
                 print_err(f"WARNING: {msg}")
             else:
                 raise ValueError(f"unknown validation mode: {self.validation_mode!r}")
+
+
+
+class NodeDataObject(NodeDataFile):
```

Comment:
I'm a bit confused by object oriented design of the node data related classes. Would it make sense to use the existing `NodeData` and move the `validate` method to that class?

Reply:
I didn't design the node-data reading classes so I am not the best to comment here, and I'm not too motivated to redesign them, but I'm happy for others to jump in here!

Reply:
[Slack thread for posterity](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1710796925576249)